sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -1036,22 +1033,17 @@ void Scheduler::GraphBuilder::decrementLeafCountersForRecord(
     MemObjRecord *Record) {
   for (Command *Cmd : Record->MReadLeaves) {
     --(Cmd->MLeafCounter);
-    if (Cmd->MLeafCounter == 0 && Cmd->isSuccessfullyEnqueued() &&
-        Cmd->supportsPostEnqueueCleanup())
+    if (Cmd->readyForCleanup())
```

Comment:
Not being familiar with the codebase, so possibly nonsense...

Any chance that having `cleanupCommandIfReady` would be a better interface now after the refactoring has happened?

Reply:
I'd rather keep it as is. One of the places where `readyForCleanup()` check is currently made is in `updateLeaves`, which cannot clean up commands on the spot (since some of its potential callers don't hold the graph lock), so we would need to keep that check regardless. In the other three cases we could go with `cleanupCommandIfReady()`, but I don't think the current version is overly verbose to warrant introducing an additional function.